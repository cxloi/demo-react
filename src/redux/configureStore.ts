import {
  // applyMiddleware,
  // compose,
  createStore,
} from 'redux';

// import monitorReducersEnhancer from "./enhancers/monitorReducers";
// import loggerMiddleware from "./middleware/logger";
import { rootReducer } from './reducers';

export default function configureStore() {
  // const middlewares = [loggerMiddleware, thunkMiddleware]
  // const middlewareEnhancer = applyMiddleware(...middlewares)

  // const enhancers = [middlewareEnhancer, monitorReducersEnhancer]
  // const composedEnhancers = compose(...enhancers)

  const store = createStore(
    rootReducer,
    // preloadedState,
    // composedEnhancers
  );

  return store;
}
