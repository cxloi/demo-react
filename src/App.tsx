import React, { CSSProperties, useEffect, useState } from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import { CustomButton } from 'demo-sb-react-components';
import { useDispatch, useSelector } from 'react-redux';
import { setConfiguration } from 'react-grid-system';
import { ThemeProvider } from '@material-ui/core/styles';
import { decreaseCountAction, increaseCountAction } from 'countModule';
import { selectCount, selectIsUserLoggedIn, selectUserDetails } from './redux/selectors';
import StoreState from './pages/StoreState';
import { userLoggedInSuccess, userLoggedOutSuccess } from './redux/actions/userActions';
import UpdateUserProfile from './pages/UpdateUserProfile';
import theme1 from './theme/theme1';
import theme2 from './theme/theme2';
import ResponsiveList from './component/ResponsiveList';

const styles: Record<string, CSSProperties> = {
  countContainer: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
  countTextContainer: {
    width: 120,
  },
  userDetailsContainer: {
    marginTop: 10,
    marginBottom: 10,
  },
};

export default function App() {
  const [theme, setTheme] = useState(theme1);

  const dispatch = useDispatch();
  const count = useSelector(selectCount);
  const isUserLoggedIn = useSelector(selectIsUserLoggedIn);
  const userDetails = useSelector(selectUserDetails);

  useEffect(() => {
    const intervalID = setInterval(() => {
      setTheme(theme === theme1 ? theme2 : theme1);
    }, 3000);
    return () => clearInterval(intervalID);
  });

  useEffect(() => {
    setConfiguration({
      maxScreenClass: 'xl',
      breakpoints: [
        theme.breakpoints.values.sm,
        theme.breakpoints.values.md,
        theme.breakpoints.values.lg,
        theme.breakpoints.values.xl,
      ],
      gutterWidth: 0,
      gridColumns: 24,
    });
  }, [theme]);

  return (
    <ThemeProvider theme={theme}>
      <Router>
        <div style={{ padding: 10 }}>
          <ul>
            <li>
              <Link to="/">Store State</Link>
            </li>
            <li>
              <Link to="/update-user-profile">Update User Profile</Link>
            </li>
          </ul>

          <div style={styles.countContainer}>
            <CustomButton label="-" onClick={() => dispatch(decreaseCountAction())} />
            <div style={styles.countTextContainer}>COUNT: {count}</div>
            <CustomButton label="+" onClick={() => dispatch(increaseCountAction())} />
          </div>

          <div style={styles.userDetailsContainer}>
            {isUserLoggedIn ? (
              <>
                <div>Name: {userDetails.userName}</div>
                <div>Age: {userDetails.age}</div>
                <CustomButton label="Log Out" onClick={() => dispatch(userLoggedOutSuccess())} />
              </>
            )
              : (
                <CustomButton
                  label="Log In"
                  onClick={() => dispatch(userLoggedInSuccess('dan-abramov-user-id'))}
                />
              )}
          </div>

          <hr />

          <Switch>
            <Route exact path="/">
              <StoreState />
            </Route>
            <Route path="/update-user-profile">
              <UpdateUserProfile />
              <ResponsiveList />
            </Route>
          </Switch>
        </div>
      </Router>
    </ThemeProvider>
  );
}
