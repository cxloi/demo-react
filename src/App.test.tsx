import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import App from './App';

const mockState = {
  count: 1,
  user: {
    isLoggedIn: true,
    userDetails: {
      userName: 'Dan Abramov',
      age: 38,
    },
  },
};
const mockStore = configureStore([])(mockState);

const componentWithStore = (
  <Provider store={mockStore}>
    <App />
  </Provider>
);
const renderComponent = () => render(componentWithStore);

describe('App', () => {
  test('"Store State" link will render store state', () => {
    const { getByTestId } = renderComponent();

    fireEvent.click(getByTestId('linkState'));
    const textElement = getByTestId('storeStatePage').textContent;
    expect(textElement).toContain('"userName": "Dan Abramov"');
  });

  test('"Update User Profile" link will render update form', () => {
    const { getByTestId } = renderComponent();

    fireEvent.click(getByTestId('linkUpdate'));
    const textElement = getByTestId('updatePage').textContent;
    expect(textElement).toContain('Update User Profile');
  });

  test('render snapshot', () => {
    const tree = renderer.create(componentWithStore).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
