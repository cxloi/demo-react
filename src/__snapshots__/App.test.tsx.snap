// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render snapshot 1`] = `
<div
  style={
    Object {
      "padding": 10,
    }
  }
>
  <ul>
    <li>
      <a
        data-testid="linkState"
        href="/"
        onClick={[Function]}
      >
        Store State
      </a>
    </li>
    <li>
      <a
        data-testid="linkUpdate"
        href="/update-user-profile"
        onClick={[Function]}
      >
        Update User Profile
      </a>
    </li>
  </ul>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
      }
    }
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-container-5 storybook-button storybook-button--medium MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        -
      </span>
    </button>
    <div
      style={
        Object {
          "width": 120,
        }
      }
    >
      COUNT: 
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-container-5 storybook-button storybook-button--medium MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        +
      </span>
    </button>
  </div>
  <div
    style={
      Object {
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    <div>
      Name: 
      Dan Abramov
    </div>
    <div>
      Age: 
      38
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-container-5 storybook-button storybook-button--medium MuiButton-containedPrimary"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "backgroundColor": undefined,
        }
      }
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Log Out
      </span>
    </button>
  </div>
  <hr />
  <div
    data-testid="updatePage"
  >
    <h2>
      Update User Profile
    </h2>
    <div>
      <div>
        User Name:
      </div>
      <input
        aria-label="input-custom-name"
        onChange={[Function]}
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 5,
          }
        }
        type="text"
        value="Dan Abramov"
      />
      <div>
        Age:
      </div>
      <input
        onChange={[Function]}
        style={
          Object {
            "marginBottom": 15,
            "marginTop": 5,
          }
        }
        type="number"
        value={38}
      />
      <br />
      <button
        className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-container-5 storybook-button storybook-button--medium MuiButton-containedPrimary"
        data-testid="updateButton"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "backgroundColor": undefined,
          }
        }
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiButton-label"
        >
          Submit
        </span>
      </button>
    </div>
  </div>
</div>
`;
